lib/api/actions/functions.php

```diff
@@ -11,83 +11,84 @@
- *                           associated with a particular action are executed. Default 10.
- *                           Lower numbers correspond with earlier execution,
- *                           and functions with the same priority are executed
- *                           in the order in which they were added to the action.
- * @param int      $args     Optional. The number of arguments the function accepts. Default 1.
- *
- * @return bool Will always return true.
+ * @since 1.5.0
+ *
+ * @param string   $id       The action's Beans ID, a unique ID tracked within Beans for this action.
+ * @param string   $hook     The name of the action to which the `$callback` is hooked.
+ * @param callable $callback The name of the function|method you wish to be called when the action event fires.
+ * @param int      $priority Optional. Used to specify the order in which the callbacks associated with a particular
+ *                           action are executed. Default is 10.
+ *                           Lower numbers correspond with earlier execution.  Callbacks with the same priority
+ *                           are executed in the order in which they were added to the action.
+ * @param int      $args     Optional. The number of arguments the callback accepts. Default is 1.
+ *
+ * @return bool
  */
 function beans_add_action( $id, $hook, $callback, $priority = 10, $args = 1 ) {
-
 	$action = array(
 		'hook'     => $hook,
 		'callback' => $callback,
 		'priority' => $priority,
 		'args'     => $args,
 	);
 
-	// Replace original if set.
-	if ( $replaced = _beans_get_action( $id, 'replaced' ) ) {
-		$action = array_merge( $action, $replaced );
+	$replaced_action = _beans_get_action( $id, 'replaced' );
+
+	// If the ID is set to be "replaced", then replace that(those) parameter(s).
+	if ( ! empty( $replaced_action ) && is_array( $replaced_action ) ) {
+		$action = array_merge( $action, $replaced_action );
 	}
 
 	$action = _beans_set_action( $id, $action, 'added', true );
 
-	// Stop here if removed.
+	// If the ID is set to be "removed", then bail out.
 	if ( _beans_get_action( $id, 'removed' ) ) {
-		return;
+		return false;
 	}
 
-	// Merge modified.
-	if ( $modified = _beans_get_action( $id, 'modified' ) ) {
-		$action = array_merge( $action, $modified );
-	}
+	$modified_action = _beans_get_action( $id, 'modified' );
 
-	// Validate action arguments.
-	if ( count( $action ) == 4 ) {
-		add_action( $action['hook'], $action['callback'], $action['priority'], $action['args'] );
+	// If the ID is set to be "modified", then modify that(those) parameter(s).
+	if ( ! empty( $modified_action ) && is_array( $modified_action ) ) {
+		$action = array_merge( $action, $modified_action );
 	}
 
-	return true;
+	// Bail out if it's not a valid action.
+	if ( ! _beans_is_action_valid( $action ) ) {
+		return false;
+	}
 
+	return add_action( $action['hook'], $action['callback'], $action['priority'], $action['args'] );
```

Comment:
Here we can return what `add_action` gives us, i.e. the `true` boolean.  One less line of code for us.